node/resolvers/Mutations/Users.ts

```diff
@@ -23,6 +25,21 @@ const addUserToMasterdata = async ({
 
   names.shift()
   const lastName = names.length > 0 ? names.join(' ') : firstName // if it gets the lastName empty, it'll repeat the firstName to avoid errors on the checkout
+
+  const userExists = await masterdata.searchDocuments({
+    dataEntity: CUSTOMER_SCHEMA_NAME,
+    fields: ['id'],
+    pagination: {
+      page: 1,
+      pageSize: 1,
+    },
+    where: `email=${email}`,
+  })
+
+  if (userExists.length) {
+    throw new UserInputError(`OrganizationUserAlreadyExists`)
+  }
```

Comment:
Here we are checking if the user exists for the current account, for any buyer org / cost center, so if this user is already registered in another buyer org, it cannot be added to any other.

The expected behavior should be: the same user cannot be registered to the same cost center in the same buyer org.
It should be possible for the same user to be part of 2 (or more) distinct cost centers in the same buyer org. It should also be possible for the user to be registered in 2 (or more) distinct buyer orgs.

Reply:
You can probably use the [getAllUsersByEmail](https://github.com/vtex-apps/storefront-permissions/blob/master/node/resolvers/Queries/Users.ts#L88) API to check if that user/email already exists for that orgId and costId

Reply:
You are right, done. Can you check again, please?